src/dask_awkward/lib/core.py

```diff
@@ -1593,6 +1593,185 @@ def _from_iter(obj):
         )
 
 
+PartialReductionType = ak.Array
+
+
+def _chunk_reducer_non_positional(
+    chunk: ak.Array | PartialReductionType,
+    is_axis_none: bool,
+    *,
+    reducer: Callable,
+    mask_identity: bool,
+) -> PartialReductionType:
+    return reducer(
```

Comment:
this function is failing with typetracer task graphs (so the necessary columns opt fails).

repro:
```python
>>> import dask
>>> import dask_awkward as dak
>>> ds = dak.from_parquet("s3://ddavistemp/hpq", storage_options={"anon": True})
>>> with dask.config.set({"awkward.optimization.on-fail": "raise"}):
...     dak.min(ds.muons.pt, axis=None).compute()
```
pretty long message, but here's the end:

```
ValueError: arrays to concatenate do not have the same depth for negative axis=-1

This error occurred while calling

    ak.concatenate(
        arrays = [<Array-typetracer [...] type='1 * 1 * ?float64'>, ('min-pre...
        axis = -1
        mergebool = True
        highlevel = True
        behavior = None
    )
```

when #277 is in we should be able to see things like this in tests :)

Also errors with `axis=0`

```
ValueError: cannot use non-negative axis on a nested list structure of variable depth (negative axis counts from the leaves of the tree; non-negative from the root)

This error occurred while calling

    ak.min(
        array = <Array-typetracer [...] type='15 * union[var * ?float64, st...'>
        axis = 0
        keepdims = True
        initial = None
        mask_identity = True
```

Reply:
Fab! I can fix this.

Reply:
Could you provide a reproducer that I can run?